Tests/SymbolKitTests/SymbolGraph/SemanticVersion/ErrorTests.swift

```diff
@@ -0,0 +1,376 @@
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":8,"minor":7,"patch":6,"buildMetadata":"z2a3b4"}"#.utf8)))
+    }
+    
+    func testLeadingZeros() {
+        //    MARK: pre-release
+        
+        assertThrowingLeadingZerosError(atPosition: .prerelease, inIdentifier: "0246", whenEvaluating: try .init(major: 1, minor: 3, patch: 5, prerelease: "0246"))
+        assertThrowingLeadingZerosError(atPosition: .prerelease, inIdentifier: "0068", whenEvaluating: try .init(major: 3, minor: 5, patch: 7, prerelease: "0068"))
+        assertThrowingLeadingZerosError(atPosition: .prerelease, inIdentifier: "0000", whenEvaluating: try .init(major: 5, minor: 7, patch: 9, prerelease: "0000"))
+        assertThrowingLeadingZerosError(atPosition: .prerelease, inIdentifier: "0654", whenEvaluating: try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":9,"minor":8,"patch":7,"prerelease":"0654"}"#.utf8)))
+        assertThrowingLeadingZerosError(atPosition: .prerelease, inIdentifier: "0065", whenEvaluating: try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":9,"minor":8,"patch":7,"prerelease":"0065"}"#.utf8)))
+        assertThrowingLeadingZerosError(atPosition: .prerelease, inIdentifier: "0000", whenEvaluating: try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":9,"minor":8,"patch":7,"prerelease":"0000"}"#.utf8)))
+        
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 2, minor: 3, patch: 4, prerelease: "0"))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":8,"minor":9,"patch":1,"prelease":"0"}"#.utf8)))
+        
+        //    MARK: build metadata
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 2, minor: 3, patch: 4, buildMetadata: "0567"))
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 8, minor: 9, patch: 1, buildMetadata: "0023"))
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 4, minor: 5, patch: 6, buildMetadata: "0000"))
+        XCTAssertNoThrow(try SymbolGraph.SemanticVersion(major: 7, minor: 8, patch: 9, buildMetadata: "0"))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":9,"minor":1,"patch":2,"buildMetadata":"0345"}"#.utf8)))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":6,"minor":7,"patch":8,"buildMetadata":"0091"}"#.utf8)))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":2,"minor":3,"patch":4,"buildMetadata":"0000"}"#.utf8)))
+        XCTAssertNoThrow(try JSONDecoder().decode(SymbolGraph.SemanticVersion.self, from: Data(#"{"major":5,"minor":6,"patch":7,"buildMetadata":"0"}"#.utf8)))
+    }
+    
+    func testOversizedNumericValues() {
+        func sum(_ summand1: String, _ summand2: String) -> Substring {
```

Comment:
I don't think it's a good idea to limit the capability of `SemanticVersion` just so it's easier to test.

Reply:
It would not limit the capability of SemanticVersion to use hardcoded strings in the test. 

For example, knowing that `UInt.max` is 18446744073709551615 (2^64-1) on a 64-bit system, it's possible to replace `String(sum("\(UInt.max)", "1")` with `"18446744073709551616" // UInt.max + 1`.

That way someone who reads the tests doesn't need to know what the `sum` function does and doesn't have wonder if the `sum` function is API. 

If looks like the assertions in `testOversizedNumericValues` only use a few values so perhaps the cleanest solution would be to define local variables for those at the start of the function
```
let largestValidNumber = "18446744073709551615" // UInt.max
let tooLargeNumber     = "18446744073709551616" // UInt.max + 1
```

Reply:
Ah I see. Sorry, I misunderstood your previous comment as adding hardcoded limits to the implementation. Thanks for the clarification!